src/engine/pieces/piece.ts

```diff
@@ -17,4 +17,15 @@ export default class Piece {
         const currentSquare = board.findPiece(this);
         board.movePiece(currentSquare, newSquare);
     }
+
+    public checkMoveConflicts(board: Board, possibleMoves: Square[], currentSquare: Square){
+        const occupiedSquares: Square[] = []
+        for (let square of possibleMoves){
+            if(board.getPiece(square) != null){
+                possibleMoves.splice(possibleMoves.indexOf(square),1)
+                occupiedSquares.push(square)
+            }
+        }
+        return [possibleMoves,occupiedSquares];
```

Comment:
I think we're adding quite a bit of complexity here by returning two arrays, can you think of any solutions where we can avoid doing this? 🙂